services/src/main/java/edu/unc/lib/dl/cdr/services/techmd/TechnicalMetadataEnhancement.java

```diff
@@ -131,45 +136,51 @@ public Element call() throws EnhancementException {
 			for (String dsid : ds2FitsDoc.keySet()) {
 				// get key PREMIS data
 				Document fits = ds2FitsDoc.get(dsid);
-				String md5checksum = fits.getRootElement().getChild("fileinfo", FITS_NS)
-						.getChildText("md5checksum", FITS_NS);
 				String size = fits.getRootElement().getChild("fileinfo", FITS_NS).getChildText("size", FITS_NS);
-
-				// IDENTIFICATION LOGIC
-				// get mimetype out of FITS XML
-				Element trustedIdentity = null;
-				Element idn = fits.getRootElement().getChild("identification", ns);
-				for (Object child : idn.getChildren("identity", ns)) {
-					Element el = (Element) child;
-					if (idn.getAttributeValue("status") == null
-							|| el.getChildren("tool", ns).size() > 1
-							|| (!"Exiftool".equals(el.getChild("tool", ns).getAttributeValue("toolname")) && !"application/x-symlink"
-									.equals(el.getAttributeValue("mimetype")))) {
-						trustedIdentity = el;
-						break;
-					}
-				}
-
-				String fitsMimetype = null;
+				
+				String fedoraMimetype = sourceMimetype.get(dsid);
+				String fitsMimetype = fedoraMimetype;
+				// Format only provided if using FITS mimetype, since otherwise it may not match
 				String format = null;
-				if (trustedIdentity != null) {
-					fitsMimetype = trustedIdentity.getAttributeValue("mimetype");
-					format = trustedIdentity.getAttributeValue("format");
-				} else {
-					format = "Unknown";
-					LOG.warn("FITS unable to conclusively identify file: {}/{}", pid, dsid);
-					LOG.info(new XMLOutputter().outputString(fits));
-				}
+				
+				// If fedora mimetype is not meaningful, then override with FITS generated value
+				if (fedoraMimetype == null || fedoraMimetype.trim().length() == 0
+						|| fedoraMimetype.contains("octet-stream")) {
+					
+					// IDENTIFICATION LOGIC
+					// get mimetype out of FITS XML
+					Element identity = null;
+					Element idn = fits.getRootElement().getChild("identification", ns);
```

Comment:
Selecting identity exclusively with an xpath doesn't seem like it is going to work without getting pretty ugly, since we can't just exclude identities that include an Exiftool entry, since FITS rolls multiple tools up into a single identity.  For example:

```xml
<fits xmlns="http://hul.harvard.edu/ois/xml/ns/fits/fits_output" xsi:schemaLocation="http://hul.harvard.edu/ois/xml/ns/fits/fits_output http://hul.harvard.edu/ois/xml/xsd/fits/fits_output.xsd" version="0.8.6" timestamp="7/14/15 2:34 PM">
  <identification>
    <identity format="Portable Document Format" mimetype="application/pdf" toolname="FITS" toolversion="0.8.6">
      <tool toolname="Jhove" toolversion="1.5" />
      <tool toolname="Tika" toolversion="1.3" />
      <tool toolname="Droid" toolversion="6.1.3" />
      <tool toolname="Exiftool" toolversion="9.13" />
      <version toolname="Jhove" toolversion="1.5">1.3</version>
      <externalIdentifier toolname="Droid" toolversion="6.1.3" type="puid">fmt/17</externalIdentifier>
    </identity>
  </identification>
....
```
wouldn't return an identity because exif is agreeing with everything else.  

I am going to stick with the loop approach and cleanup the status evaluation bit.  We can revisit our approach to deciding on what is the preferred FITS mimetype later if we are still having issues.